src/stackwalk.c

```diff
@@ -1108,6 +1108,42 @@ JL_DLLEXPORT void jl_print_backtrace(void) JL_NOTSAFEPOINT
     jlbacktrace();
 }
 
+JL_DLLEXPORT void jl_print_task_backtraces(void)
+{
+    jl_task_t *t;
+    jl_ptls_t ptls;
+    arraylist_t *live_tasks;
+    size_t i, j, n;
+
+    for (i = 0; i < jl_n_threads; i++) {
+        ptls = jl_all_tls_states[i];
+        live_tasks = &ptls->heap.live_tasks;
+        n = live_tasks->len;
+        jl_safe_printf("==== Thread %d (%zu live tasks)\n",
+                ptls->tid + 1, n + 1);
```

Comment:
Heh, i've already forgotten again since JuliaCon why exactly this array is per-thread, since tasks can of course move threads, and when the task is blocking, it's not scheduled on any thread...

Does this correlate to which Thread _created_ the task? Could we maybe clarify the relationship in the println? (For example `==== Tasks created by Thread %d` or something like that?)

Reply:
A task is put onto `ptls->heap.live_tasks` [here](https://github.com/JuliaLang/julia/blob/master/src/gc-stacks.c#L179), when a stack is allocated for it. This usually happens when in the creating task's context and hence on the thread that was running the creating task but in some situations, this can happen when the created task first runs, which could be on any thread.

From a point of view, this partitioning of tasks across threads is not meaningful. But I can add that clarification in the output anyway.

Reply:
Thanks, yeah i think that will be helpful!